src/xo-app/host/tab-network.js

```diff
@@ -117,57 +128,143 @@ class PifItem extends Component {
+        },
+        noop
+      )
+    }
+    return reconfigurePifIp(this.props.pif, { mode })
+  }
 
   render () {
-    const { networks, pif, vifsByNetwork } = this.props
+    const {pif} = this.props
     const { configModes } = this.state
+    return <Select
+      onChange={this._configIp}
+      options={configModes}
+      value={pif.mode}
+    >
+      {pif.mode}
+    </Select>
+  }
+}
 
+@connectStore(() => ({
+  vifsByNetwork: createGetObjectsOfType('VIF').groupBy('$network')
+}))
+class PifItemInUse extends Component {
+  render () {
+    const {networks, pif, vifsByNetwork} = this.props
     const pifInUse = some(vifsByNetwork[pif.$network], vif => vif.attached)
+    return <div className='text-xs-center'>
+      {_toggleDefaultLockingMode(
+        <Toggle
+          disabled={pifInUse}
+          onChange={() => editNetwork(pif.$network, { defaultIsLocked: !networks[pif.$network].defaultIsLocked })}
+          value={networks[pif.$network].defaultIsLocked}
+        />,
+        pifInUse && _('pifInUse')
+      )}
+    </div>
+  }
+}
 
-    return <tr>
-      <td>{pif.device}</td>
-      <td>{networks[pif.$network].name_label}</td>
-      <td>
-        {pif.vlan === -1
-          ? 'None'
-          : <Number value={pif.vlan} onChange={this._editPif}>
-            {pif.vlan}
-          </Number>
-        }
-      </td>
-      <td>
-        {pif.ip}
-        {' '}
-        {pif.ip && <a className='hidden-md-down' onClick={this._onEditIp} style={EDIT_BUTTON_STYLE}>
-          <Icon icon='edit' size='1' fixedWidth />
-        </a>}
-      </td>
-      <td>
-        <Select
-          onChange={this._configIp}
-          options={configModes}
-          value={pif.mode}
-        >
-          {pif.mode}
-        </Select>
-      </td>
-      <td><pre>{pif.mac}</pre></td>
-      <td>{pif.mtu}</td>
-      <td className='text-xs-center'>
-        {_toggleDefaultLockingMode(
-          <Toggle
-            disabled={pifInUse}
-            onChange={() => editNetwork(pif.$network, { defaultIsLocked: !networks[pif.$network].defaultIsLocked })}
-            value={networks[pif.$network].defaultIsLocked}
-          />,
-          pifInUse && _('pifInUse')
-        )}
-      </td>
-      <td>
+const COLUMNS = [
+  {
+    itemRenderer: (pif, networks) => {
+      return pif.device
+    },
+    default: true,
+    name: _('pifDeviceLabel'),
+    sortCriteria: 'device'
+  },
+  {
+    itemRenderer: (pif, networks) => {
+      return networks[pif.$network].name_label
+    },
+    name: _('pifNetworkLabel'),
+    sortCriteria: (pif, networks) => networks[pif.$network].name_label
+  },
+  {
+    itemRenderer: (pif, networks) => {
+      return <PifItemVlan key={pif.id} pif={pif} networks={networks} />
+    },
+    name: _('pifVlanLabel'),
+    sortCriteria: 'vlan'
+  },
+  {
+    itemRenderer: (pif, networks) => {
+      return <PifItemIp key={pif.id} pif={pif} networks={networks} />
```

Comment:
Remove `key` prop. Check [this](https://reactjs.org/docs/lists-and-keys.html#keys) to know what's the keys purpose and thus why you don't need it here.

Reply:
I've create key prop for switch function that i've created in the first big component in order to identify which one to render and i forgot to remove it. Now it's clean :)